relationship.go

```diff
@@ -0,0 +1,53 @@
+package jsh
+
+import (
+	"fmt"
+
+	"encoding/json"
+)
+
+// Relationship represents a reference from the resource object in which it's
+// defined to other resource objects.
+type Relationship struct {
+	Links *Links                 `json:"links,omitempty"`
+	Data  ResourceLinkage        `json:"data,omitempty"`
+	Meta  map[string]interface{} `json:"meta,omitempty"`
+}
+
+// ResourceLinkage is a typedef around a slice of resource identifiers. This
+// allows us to implement a custom UnmarshalJSON.
+type ResourceLinkage []*ResourceIdentifier
+
+// ResourceIdentifier identifies an individual resource.
+type ResourceIdentifier struct {
+	Type string `json:"type" valid:"required"`
+	ID   string `json:"id" valid:"required"`
+}
+
+/*
+UnmarshalJSON allows us to manually decode a the resource linkage via the
+json.Unmarshaler interface.
+*/
+func (ri *ResourceLinkage) UnmarshalJSON(data []byte) error {
```

Comment:
Minor nitpick, but can you update `ri` to be `r`, `rl`, `resource` or `linkage`. Unless I am missing something with the naming convention.